src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts

```diff
@@ -315,18 +344,18 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
 		}
 	}
 
-	private _guessProfileIcon(shell: string): string | undefined {
+	private _guessProfileIcon(shell: string): ThemeIcon | undefined {
 		const file = path.parse(shell).name;
 		switch (file) {
 			case 'bash':
-				return Codicon.terminalBash.id;
+				return { id: Codicon.terminalBash.id };
```

Comment:
Probably worth doing a search for `{ id: Codicon` to catch other cases of creating objects where we don't need to